### PR TITLE
Refactor/online import

### LINKED ISSUE #1146: bug: after sync stream features, the value of oomcli get online is empty
To reproduce:

```shell
$ oomcli init
$ oomcli apply -f /tmp/tmp/meta.yaml
applied
$ oomcli get meta entity -o yaml
items:
  - kind: Entity
    name: user
    description: user
    groups:
      - name: user-click
        category: stream
        snapshot-interval: 1s
        description: user click post feature
        features:
          - name: last_5_click_posts
            value-type: string
            description: user last 5 click posts
          - name: number_of_user_starred_posts
            value-type: int64
            description: number of posts that users starred today
  - kind: Entity
    name: device
    description: device
    groups:
      - name: phone
        category: batch
        description: phone
        features:
          - name: price
            value-type: int64
            description: price
          - name: model
            value-type: string
            description: model
$ cat user_click.csv
user,unix_milli,last_5_click_posts,number_of_user_starred_posts
1,1644285708319,"1,2",10
2,1644285708319,"2,3",10
$ oomcli import --group user-click --input-file user_click.csv --description 'test data'
importing features ...
succeeded
RevisionID: 0
$ oomcli sync --group-name user-click
syncing features ...
INSERT INTO "tmp_online_stream_1644312413532942845" ("user","last_5_click_posts","number_of_user_starred_posts") VALUES (?, ?, ?),(?, ?, ?)
1
1,2
10
2
2,3
10
succeeded.
$ oomcli get online -k 1 --feature user-click.last_5_click_posts
+------+-------------------------------+
| user | user-click.last_5_click_posts |
+------+-------------------------------+
|    1 |                               |
+------+-------------------------------+
$ litecli /tmp/tmp/oomstore.db
Version: 1.7.0
Mail: https://groups.google.com/forum/#!forum/litecli-users
GitHub: https://github.com/dbcli/litecli
/tmp/tmp/oomstore.db> .tables ;
+-----------------------------------------+
| name                                    |
+-----------------------------------------+
| entity                                  |
| feature                                 |
| feature_group                           |
| feature_group_revision                  |
| offline_stream_cdc_1_0                  |
| offline_stream_cdc_1_1644285708319      |
| offline_stream_snapshot_1_0             |
| offline_stream_snapshot_1_1644285708319 |
| online_stream_1                         |
| tmp_export_entity_1644312413504106444   |
| tmp_online_stream_1644312413532942845   |
+-----------------------------------------+
Time: 0.007s
/tmp/tmp/oomstore.db> select * from online_stream_1 ;
0 rows in set
Time: 0.000s
/tmp/tmp/oomstore.db> select * from tmp_online_stream_1644312413532942845 ;
+------+--------------------+------------------------------+
| user | last_5_click_posts | number_of_user_starred_posts |
+------+--------------------+------------------------------+
| 1    | 1,2                | 10                           |
| 2    | 2,3                | 10                           |
+------+--------------------+------------------------------+
2 rows in set
Time: 0.004s
/tmp/tmp/oomstore.db>
```

### PR DESCRIPTION
**Pull request by lianxmfor**
this PR does:
1. `types.ExportRecord` add new field `Error error`
2. `types.ExportResult` remove field `error <- error`

fix #1146 

note: This PR is just one way to do it, and whether or not to do it is up for discussion.

### CODE REVIEW

File: pkg/oomstore/types/types.go [hunk 1]
```diff
@@ -15,14 +15,17 @@ const (
 	TableStreamCdc      TableType = "stream_cdc"
 )
 
-type ExportRecord []interface{}
+type ExportRecord struct {
+	Records []interface{}
```
Comment (wfxr) [inline pkg/oomstore/types/types.go:19]: Records -> Record

**Review by wfxr — APPROVED** [review]
lgtm

File: pkg/oomstore/types/types.go (hunk 1 above)
Comment (lianxmfor) [reply pkg/oomstore/types/types.go]: done

File: internal/database/online/tikv/import.go [hunk 2]
```diff
@@ -41,11 +41,15 @@ func (db *DB) Import(ctx context.Context, opt online.ImportOpt) error {
 	var putVals [][]byte
 
 	for record := range opt.ExportStream {
-		if len(record) != len(opt.Features)+1 {
-			return errdefs.Errorf("field count not matched, expected %d, got %d", len(opt.Features)+1, len(record))
+		if record.Error != nil {
+			return err
```
Comment (YikSanChan) [inline internal/database/online/tikv/import.go:45]: should it be record.Error?

File: pkg/oomstore/export.go [hunk 3]
```diff
@@ -120,9 +120,13 @@ func (s *OomStore) Export(ctx context.Context, opt types.ExportOpt) error {
 		return err
 	}
 	for row := range exportResult.Data {
-		if err := w.Write(cast.ToStringSlice([]interface{}(row))); err != nil {
+		if row.Error != nil {
+			return err
```
Comment (YikSanChan) [inline pkg/oomstore/export.go:124]: row.Error?

File: internal/database/online/tikv/import.go (hunk 2 above)
Comment (lianxmfor) [reply internal/database/online/tikv/import.go]: good catch! fixed

File: pkg/oomstore/export.go (hunk 3 above)
Comment (lianxmfor) [reply pkg/oomstore/export.go]: fixed

**Review by YikSanChan — APPROVED** [review]
lgtm

**Pull request merged**